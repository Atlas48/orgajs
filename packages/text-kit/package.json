{
  "name": "text-kit",
  "version": "2.3.16",
  "description": "bundle of useful tools for manipulating text",
  "author": "Xiaoxing Hu <hi@huxiaoxing.com>",
  "homepage": "https://github.com/orgapp/orgajs/tree/master/packages/text-kit#readme",
  "repository": {
    "type": "git",
    "url": "https://github.com/orgapp/orgajs.git",
    "directory": "packages/text-kit"
  },
  "license": "MIT",
  "main": "dist/index.js",
  "directories": {
    "dist": "dist",
    "test": "__tests__"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "precompile": "yarn clean",
    "compile": "tsc",
    "prepare": "yarn compile",
    "watch": "yarn run compile -w",
    "clean": "del ./dist/*"
  },
  "bugs": {
    "url": "https://github.com/orgapp/orgajs/issues"
  }
}
