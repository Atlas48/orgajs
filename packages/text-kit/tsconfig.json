{
  "compilerOptions": {
    "outDir": "./lib",
    "module": "commonjs",
    "noImplicitAny": false,
    "removeComments": true,
    "noLib": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es6",
    "sourceMap": true,
    "moduleResolution": "node",
    "skipLibCheck": true,
    "noImplicitUseStrict": true,
    "noImplicitReturns": true,
    "esModuleInterop": true,
    "declaration": true,
    "lib": [
      "es7", "es2019"
    ]
  },
  "include": ["./src"],
  "exclude": [
    "node_modules",
    "**/*.spec.ts"
  ]

}
