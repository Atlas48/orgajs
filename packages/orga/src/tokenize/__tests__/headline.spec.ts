import { read } from "text-kit";
import { tokenize } from "../index";

describe("tokenize headline", () => {
  const tok = (text: string) => {
    const { substring } = read(text);
    const tokens = tokenize(text).all();
    return tokens.map(({ position, ...rest }) => ({
      ...rest,
      _content: substring(position)
    }));
  };

  it("knows headlines", () => {
    expect(tok("** a headline")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "**",
          "level": 2,
          "type": "stars",
        },
        Object {
          "_content": "a headline",
          "type": "text.plain",
          "value": "a headline",
        },
      ]
    `);

    expect(tok("** _headline_")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "**",
          "level": 2,
          "type": "stars",
        },
        Object {
          "_content": "_headline_",
          "type": "text.underline",
          "value": "headline",
        },
      ]
    `);

    expect(tok("**   a headline")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "**",
          "level": 2,
          "type": "stars",
        },
        Object {
          "_content": "a headline",
          "type": "text.plain",
          "value": "a headline",
        },
      ]
    `);

    expect(tok("***** a headline")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "*****",
          "level": 5,
          "type": "stars",
        },
        Object {
          "_content": "a headline",
          "type": "text.plain",
          "value": "a headline",
        },
      ]
    `);

    expect(tok("* a ðŸ˜€line")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "*",
          "level": 1,
          "type": "stars",
        },
        Object {
          "_content": "a ðŸ˜€line",
          "type": "text.plain",
          "value": "a ðŸ˜€line",
        },
      ]
    `);

    expect(tok("* TODO [#A] a headline     :tag1:tag2:"))
      .toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "*",
          "level": 1,
          "type": "stars",
        },
        Object {
          "_content": "TODO",
          "actionable": true,
          "keyword": "TODO",
          "type": "todo",
        },
        Object {
          "_content": "[#A]",
          "type": "priority",
          "value": "[#A]",
        },
        Object {
          "_content": "a headline",
          "type": "text.plain",
          "value": "a headline",
        },
        Object {
          "_content": ":tag1:tag2:",
          "tags": Array [
            "tag1",
            "tag2",
          ],
          "type": "tags",
        },
      ]
    `);

    expect(tok("* TODO [#A] a headline     :tag1:tag2:"))
      .toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "*",
          "level": 1,
          "type": "stars",
        },
        Object {
          "_content": "TODO",
          "actionable": true,
          "keyword": "TODO",
          "type": "todo",
        },
        Object {
          "_content": "[#A]",
          "type": "priority",
          "value": "[#A]",
        },
        Object {
          "_content": "a headline",
          "type": "text.plain",
          "value": "a headline",
        },
        Object {
          "_content": ":tag1:tag2:",
          "tags": Array [
            "tag1",
            "tag2",
          ],
          "type": "tags",
        },
      ]
    `);
  });

  it("knows these are not headlines", () => {
    expect(tok("*not a headline")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "*not a headline",
          "type": "text.plain",
          "value": "*not a headline",
        },
      ]
    `);

    expect(tok(" * not a headline")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "* not a headline",
          "type": "text.plain",
          "value": "* not a headline",
        },
      ]
    `);
    expect(tok("*_* not a headline")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "*_*",
          "type": "text.bold",
          "value": "_",
        },
        Object {
          "_content": " not a headline",
          "type": "text.plain",
          "value": " not a headline",
        },
      ]
    `);
    expect(tok("not a headline")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_content": "not a headline",
          "type": "text.plain",
          "value": "not a headline",
        },
      ]
    `);
  });
});
