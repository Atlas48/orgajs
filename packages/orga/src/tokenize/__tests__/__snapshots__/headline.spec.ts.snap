// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize headline knows headlines 1`] = `
Array [
  Object {
    "data": Object {
      "level": 2,
    },
    "level": 2,
    "position": Object {
      "end": Object {
        "column": 2,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "stars",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 3,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize headline knows headlines 2`] = `
Array [
  Object {
    "data": Object {
      "level": 2,
    },
    "level": 2,
    "position": Object {
      "end": Object {
        "column": 2,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "stars",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 3,
        "line": 0,
      },
    },
    "type": "text.underline",
  },
]
`;

exports[`tokenize headline knows headlines 3`] = `
Array [
  Object {
    "data": Object {
      "level": 2,
    },
    "level": 2,
    "position": Object {
      "end": Object {
        "column": 2,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "stars",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 15,
        "line": 0,
      },
      "start": Object {
        "column": 5,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize headline knows headlines 4`] = `
Array [
  Object {
    "data": Object {
      "level": 5,
    },
    "level": 5,
    "position": Object {
      "end": Object {
        "column": 5,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "stars",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 16,
        "line": 0,
      },
      "start": Object {
        "column": 6,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize headline knows headlines 5`] = `
Array [
  Object {
    "data": Object {
      "level": 1,
    },
    "level": 1,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "stars",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 10,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize headline knows headlines 6`] = `
Array [
  Object {
    "data": Object {
      "level": 1,
    },
    "level": 1,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "stars",
  },
  Object {
    "actionable": true,
    "keyword": "TODO",
    "position": Object {
      "end": Object {
        "column": 6,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "todo",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 7,
        "line": 0,
      },
    },
    "type": "priority",
    "value": "[#A]",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 22,
        "line": 0,
      },
      "start": Object {
        "column": 12,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 38,
        "line": 0,
      },
      "start": Object {
        "column": 27,
        "line": 0,
      },
    },
    "tags": Array [
      "",
      "tag1",
      "tag2",
      "",
    ],
    "type": "tags",
  },
]
`;

exports[`tokenize headline knows these are not headlines 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 15,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize headline knows these are not headlines 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 17,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize headline knows these are not headlines 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 3,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.bold",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 18,
        "line": 0,
      },
      "start": Object {
        "column": 3,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize headline knows these are not headlines 4`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 14,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;
